### PR TITLE
Add Postgres tasks

### PR DESCRIPTION
**Pull request by zangell44**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?

This PR adds two new tasks for interacting with Postgres databases, `PostgresExecute` and `PostgresFetch`, which execute and fetch results of a SQL query.

Basic tests have been added for both tasks. Functionality has been tested locally with a live Postgres instance.

## Why is this PR important?

Closes [#1022](https://github.com/PrefectHQ/prefect/issues/1022).

### CODE REVIEW

File: src/prefect/tasks/postgres/postgres.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+class ExecuteTask(Task):
+    """
+        Task for executing a query against a Postgres database.
+
+        Args:
+            - db_name (str):
+            - user (str):
+            - password (str):
+            - host (str):
```
Comment (cicdw) [inline src/prefect/tasks/postgres/postgres.py]: Forgotten arg docs?

File: src/prefect/tasks/postgres/postgres.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+class FetchTask(Task):
+    """
+        Task for fetching results of query from Postgres database.
+
+        Args:
+            - db_name (str):
+            - user (str):
+            - password (str):
+            - host (str):
```
Comment (cicdw) [inline src/prefect/tasks/postgres/postgres.py]: Same as above

File: src/prefect/tasks/postgres/postgres.py (hunk 1 above)
Comment (zangell44) [reply src/prefect/tasks/postgres/postgres.py]: Damnit haha I can't believe I forgot those

**Comment by zangell44** [comment]
@cicdw yes, I can add that markdown file.

I noticed that the API References in the current ones link to 404 pages. Should I follow the same format for the links and expect those pages to be functional?

**Comment by cicdw** [comment]
@zangell44 where are you seeing 404 errors?  Ultimately those links get deployed here: https://docs.prefect.io/guide/task_library/aws.html (for example) and these links should work.

**Comment by zangell44** [comment]
@cicdw gotcha, I was seeing 404 errors from when you click on the API references directly in the markdown files here https://github.com/PrefectHQ/prefect/blob/master/docs/guide/task_library/aws.md

I'll push changes to add a markdown file in a bit. 

I also came across the SQLite tasks and I think it's better to rename the ones I created to something like `PostgresExecute` and `PostgresFetch` to be more consistent/descriptive.

File: src/prefect/tasks/postgres/postgres.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        Args:
+            - query (str, optional): query to execute against database
+            - data (tuple, optional): values to use in query, must be specified using
+                placeholder is query string
+            - commit (bool, optional): set to True to commit transaction, defaults to false
+
+        Returns:
+            - None
+        """
+        if query == "":
```
Comment (joshmeek) [inline src/prefect/tasks/postgres/postgres.py]: Let's default the `query` to None and change this to `if not query` so it'll catch the cases where it's None and ""

File: src/prefect/tasks/postgres/postgres.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+        Args:
+            - query (str, optional): query to execute against database
+            - data (tuple, optional): values to use in query, must be specified using
+                placeholder is query string
+            - commit (bool, optional): set to True to commit transaction, defaults to false
+
+        Returns:
+            - None
+        """
```
Comment (joshmeek) [inline src/prefect/tasks/postgres/postgres.py]: Can you add to the docstring

```
Raises:
	- DatabaseError: ..why it would be raised...
```

Comment (joshmeek) [reply src/prefect/tasks/postgres/postgres.py]: Not sure what to do about the base `Exception` raise option though. Does that needs to be documented @cicdw?

File: src/prefect/tasks/postgres/postgres.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+            - fetch (str, optional): one | many | all, used to determine how many results to fetch from executed query
+            - fetch_count (int, optional): if fetch = 'many', determines the number of results to fetch, defaults to 10
+            - query (str, optional): query to execute against database
+            - data (tuple, optional): values to use in query, must be specified using placeholder is query string
+            - commit (bool, optional): set to True to commit transaction, defaults to false
+
+        Returns:
+            - records (tuple or list of tuples): records from provided query
+        """
+        if query == "":
```
Comment (joshmeek) [inline src/prefect/tasks/postgres/postgres.py]: Same query comment as the other task

File: src/prefect/tasks/postgres/postgres.py (hunk 4 above)
Comment (joshmeek) [reply src/prefect/tasks/postgres/postgres.py]: Oh, can you also add one for the `ValueError` as well?

File: src/prefect/tasks/postgres/postgres.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        Args:
+            - fetch (str, optional): one | many | all, used to determine how many results to fetch from executed query
+            - fetch_count (int, optional): if fetch = 'many', determines the number of results to fetch, defaults to 10
+            - query (str, optional): query to execute against database
+            - data (tuple, optional): values to use in query, must be specified using placeholder is query string
+            - commit (bool, optional): set to True to commit transaction, defaults to false
+
+        Returns:
+            - records (tuple or list of tuples): records from provided query
+        """
```
Comment (joshmeek) [inline src/prefect/tasks/postgres/postgres.py]: Same `Raises` docstring requests as the other task

File: src/prefect/tasks/postgres/postgres.py (hunk 4 above)
Comment (cicdw) [reply src/prefect/tasks/postgres/postgres.py]: I don't think the base `Exception` needs to be documented; I view the `Raises` section as the errors that one might "expect" to see when running these tasks and those that are explicitly raised by the task.

**Comment by zangell44** [comment]
@joshmeek I added the exceptions and modified the logic for the query parameter in the latest commits. Let me know if there's anything else I can do on this.

Also, a quick GitHub etiquette thing - would you prefer I "Resolve conversation" after making requested edits, or leave the conversation until you review changes?

**Comment by joshmeek** [comment]
@zangell44 Looks great! I'm all fine with marking the convos as resolved once you implement them

**Comment by cicdw** [comment]
I prematurely merged my own PR and didn't think about the merge conflict this would cause, my apologies!!

**Pull request merged**